scripts/get-modules-data.ts

```diff
@@ -0,0 +1,76 @@
+import path from "path";
+import fs from "fs";
+import fsp from "fs/promises";
+import { fileURLToPath } from "url";
+import { PackageJson } from "type-fest";
+import { isNonNullable } from "./utils";
+
+export type ModulePkg = PackageJson & {
+  primitive?: {
+    list?: string[];
+    category?: string;
+    stage?: number;
+  };
+};
+
+export type ModuleData = {
+  name: string;
+  version: string;
+  description: string;
+  tags: string[];
+  category: string;
+  stage: number;
+  primitives: string[];
+  localDependencies: string[];
+  peerDependencies: string[];
+};
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const rootPath = path.join(__dirname, "..");
+const packagesPath = path.join(rootPath, "packages");
+
+export async function getModulesData(): Promise<ModuleData[]>;
+export async function getModulesData<T>(mapFn: (data: ModuleData) => T | Promise<T>): Promise<T[]>;
+export async function getModulesData<T = ModuleData>(
+  mapFn: (data: ModuleData) => T = moduleData => moduleData as unknown as T,
+): Promise<T[]> {
+  const packageNames = await fsp.readdir(packagesPath);
+
+  const promises = packageNames.map(async name => {
+    const packageJsonPath = path.join(packagesPath, name, "package.json");
+    if (!fs.existsSync(packageJsonPath)) {
+      // eslint-disable-next-line no-console
+      console.warn(`package ${name} doesn't have package.json`);
+      return null;
+    }
+
+    const pkg = JSON.parse(await fsp.readFile(packageJsonPath, "utf8")) as ModulePkg;
+    if (!pkg.primitive) {
+      // eslint-disable-next-line no-console
+      console.warn(`package ${name} doesn't have primitive field in package.json`);
+      return null;
+    }
+
+    const dependencies = Object.keys(pkg.dependencies ?? {});
+    const peerDependencies = Object.keys(pkg.peerDependencies ?? {});
+    const localDependencies = dependencies.filter(dep => dep.startsWith("@solid-primitives/"));
+
+    return {
+      data: await mapFn({
+        name,
+        version: pkg.version ?? "0.0.0",
+        description: pkg.description ?? "",
+        tags: pkg.keywords ?? [],
+        category: pkg.primitive.category ?? "Misc",
+        stage: pkg.primitive.stage ?? 0,
+        primitives: pkg.primitive.list ?? [],
```

Comment:
You're getting primitive list from package.json file, it contains missing  and removed primitives. Shouldn't we get primitives from source like here https://github.com/solidjs-community/solid-primitives/blob/main/scripts/update-site/get-package-data.ts#L82-L144

Reply:
The list in package.json is the source of truth of what are the main primitives for given package. Only those should be displayed in the primitives table.
If a list contains missing primitives, it could just be fixed. And if it misses others, it may be on purpose.

Reply:
> The list in package.json is the source of truth of what are the main primitives for given package. Only those should be displayed in the primitives table just like it works on the main readme.

I agree with primitives table just showing main primitives.

> But I was thinking about the possibility of adding the rest of the exports to a additionalExports field, so that they could be used by tooltips, or displayed in some way in the package page in the future.

I would like additionalExports field to be added, because I found so many helpful primitives that the README Table didn't provide, when I was building/testing the Search function. It would be nice for users to be able to discover non-main primitives. If it's too much noise the Search function, I could put it behind a toggle.